yorick.mli

```diff
@@ -0,0 +1,88 @@
+(*
+ * Copyright (c) 2015 David Sheets <sheets@alum.mit.edu>
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ *
+ *)
+
+(** [tokens *~ sep] is the string of [tokens] separated by [sep]. *)
+val ( *~ ) : string list -> string -> string
+
+(** [?|  command] executes [command] successfully. *)
+val ( ?|  ) : string -> unit
+
+(** [?|. command_pattern] is a function of [command_pattern] parameters
+   which executes [command_pattern ...] successfully. *)
+val ( ?|. ) : ('a, unit, string, unit) format4 -> 'a
+
+(** [?|~ command_pattern] is a function of [command_pattern] parameters
+   which prints [command_pattern ...] and then executes it successfully. *)
+val ( ?|~ ) : ('a, unit, string, unit) format4 -> 'a
+
+(** [?|> command_pattern] is a function of [command_pattern]
+    parameters which returns a string of the stdout from
+    [command_pattern ...]'s execution. *)
+val ( ?|> ) : ('a, unit, string, string) format4 -> 'a
+
+(** [?|? command_pattern] is a function of [command_pattern]
+    parameters which returns an int of the exit code from
+    [command_pattern ...]'s execution. *)
+val ( ?|? ) : ('a, unit, string, int) format4 -> 'a
+
```

Comment:
It might be better (in the context of other uses) to have the infix operators in their own submodule so that it can be opened without pulling in functions like `list` into the current module namespace